PIC_database_updater.py

```diff
@@ -167,112 +272,109 @@ def update_TRS(self, barcode, township:str, range:str, section:str):
 
-        sql = self.sql_csv_tools.create_update_statement(tab_name='localitydetail',
-                                                         col_list=['Township', 'RangeDesc', 'Section'],
-                                                         val_list=[township, range, section], condition=condition)
+        else:
+            if 'Township' and 'Range' and 'Section' in self.update_frame:
 
-        self.sql_csv_tools.insert_table_record(sql=sql)
+                self.update_trs(locality_det_id=locality_det_id, row=row)
 
-    def update_UTM(self, barcode, easting: float, northing: float, datum: str):
-        """update_utm: updates utm fields on database
-            args:
-                barcode: the barcode of the database record to update.
-                easting: utm easting
-                northing: utm northing
-                datum: utm datum"""
+            else:
+                self.logger.info(f"No TRS Fields in data, skipping update")
 
-        collecting_event_id = self.get_collectingevent_id(barcode=barcode)
+            if 'UtmNorthing' and 'UtmEasting' and 'UtmDatum' in self.update_frame:
 
-        locality_id = self.get_locality_id_with_collectingevent(collecting_event_id=collecting_event_id)
+                self.update_utm(locality_det_id=locality_det_id, row=row)
 
-        locality_det_id = self.sql_csv_tools.get_one_match(tab_name='localitydetail',
-                                                           id_col='LocalityDetailID',
-                                                           key_col='LocalityID', match=locality_id,
-                                                           match_type=int)
+            else:
+                self.logger.info(f"No UTM fields in data, skipping update")
 
-        condition = f"""WHERE LocalityDetailID = {locality_det_id};"""
+    def update_trs(self, locality_det_id, row):
+        """update_trs: updates TRS fields on database table localitydetail
+            args:
+                locality_det_id: the localitydetail ID to update.
+                row: row from update csv"""
 
-        sql = self.sql_csv_tools.create_update_statement(tab_name='localitydetail',
-                                                         col_list=['UtmEasting', 'UtmNorthing', 'UtmDatum'],
-                                                         val_list=[easting, northing, datum], condition=condition)
+        is_present = self.sql_csv_tools.get_one_match(tab_name="localitydetail", id_col="Township",
+                                                      key_col="LocalityDetailID",
+                                                      match=f"{locality_det_id}", match_type=int)
 
-        self.sql_csv_tools.insert_table_record(sql=sql)
+        if is_present or self.force_update:
 
+            condition = f"""WHERE LocalityDetailID = {locality_det_id};"""
 
-    # leaving this incomplete/ commented as unsure about final structure of determiner data
```

Comment:
removed the update determiner function, as we already decided we were not going to transcribe this field, and just leave it as info on the specimen sheet image.